src/server/game/Entities/Player/Player.cpp

```diff
@@ -27553,3 +27605,228 @@ void Player::SummonPet(uint32 entry, float x, float y, float z, float ang, PetTy
+                diffz > 1.87f &&
+                tanangle < 0.57735026919f) // 30 degrees
+            {
+                std::string mapname = GetMap()->GetMapName();
+                sLog->outString("CheckMovementInfo :  Climb-Hack detected for Account id : %u, Player %s, diffZ = %f, distance = %f, angle = %f, Map = %s, mapId = %u, X = %f, Y = %f, Z = %f",
+                    GetSession()->GetAccountId(), GetName().c_str(), diffz, distance, tanangle, mapname.c_str(), GetMapId(), GetPositionX(), GetPositionY(), GetPositionZ());
+                sWorld->SendGMText(LANG_GM_ANNOUNCE_WALLCLIMB, GetSession()->GetAccountId(), GetName().c_str(), diffz, distance, tanangle, mapname.c_str(), GetMapId(), GetPositionX(), GetPositionY(), GetPositionZ());
+                return false;
+            }
+        }
+
+        uint32 oldstime = GetLastMoveServerTimestamp();
+        uint32 stime = getMSTime();
+        uint32 ping, latency;
+        movetime = movementInfo.time;
+        latency = GetSession()->GetLatency();
+        ping = std::max(uint32(60), latency);
+
+        speed = GetSpeed(MOVE_RUN);
+        if (isSwimming())
+            speed = GetSpeed(MOVE_SWIM);
+        if (IsFlying() || this->CanFly())
+            speed = GetSpeed(MOVE_FLIGHT);
+
+        delaysentrecieve = (ctime - oldstime) / 10000000000;
+        delay = fabsf(movetime - stime) / 10000000000 + delaysentrecieve;
+        difftime = (movetime - ctime + ping) * 0.001f + delay;
+        normaldistance = speed * difftime; // if movetime faked and lower, difftime should be with "-"
+
+        if (distance < normaldistance)
+            return true;
+
+        GetPosition(x, y);
+
+        sLog->outString("CheckMovementInfo :  SpeedHack Detected for Account id : %u, Player %s", GetSession()->GetAccountId(), GetName().c_str());
+        sLog->outString("========================================================");
+        sLog->outString("CheckMovementInfo :  oldX = %f", x);
+        sLog->outString("CheckMovementInfo :  oldY = %f", y);
+        sLog->outString("CheckMovementInfo :  newX = %f", npos.GetPositionX());
+        sLog->outString("CheckMovementInfo :  newY = %f", npos.GetPositionY());
+        sLog->outString("CheckMovementInfo :  packetdistance = %f", distance);
+        sLog->outString("CheckMovementInfo :  available distance = %f", normaldistance);
+        sLog->outString("CheckMovementInfo :  movetime = %f", movetime);
+        sLog->outString("CheckMovementInfo :  delay sent ptk - recieve pkt (previous) = %f", delaysentrecieve);
+        sLog->outString("CheckMovementInfo :  FullDelay = %f", delay);
+        sLog->outString("CheckMovementInfo :  difftime = %f", difftime);
+        sLog->outString("CheckMovementInfo :  latency = %u", latency);
+        sLog->outString("CheckMovementInfo :  ping = %u", ping);
+
+        sWorld->SendGMText(LANG_GM_ANNOUNCE_ASH, GetName().c_str(), normaldistance, distance);
+    }
+    else
+        return true;
+
+    return false;
+}
```

Comment:
@kvipka the file Player.cpp is already **HUGE**, would it be possible to wrap this logic somewhere else? my IDE even refuses to scan it since it's so big.

Reply:
@FrancescoBorzi yes i can to create a new .cpp file and move this func there.

Reply:
@kvipka that would be great, thank you